src/hooks/useQuota/useQuota.tsx

```diff
@@ -0,0 +1,143 @@
+    transform(
+      quota.globalQuota!,
+      (result: Quota, itemQuota) => {
+        if (products.some(policy => policy.category === itemQuota.category))
+          result.globalQuota!.push(itemQuota);
+      },
+      filteredQuota
+    );
+  }
+
+  if (quota.localQuota) {
+    filteredQuota.localQuota = [];
+    transform(
+      quota.localQuota!,
+      (result: Quota, itemQuota) => {
+        if (products.some(policy => policy.category === itemQuota.category))
+          result.localQuota!.push(itemQuota);
+      },
+      filteredQuota
+    );
+  }
+
+  return filteredQuota;
+};
+
+const hasNoQuota = (quota: Quota): boolean => {
+  if (quota === null) {
+    return true;
+  }
+  return quota.remainingQuota.every(item => item.quantity === 0);
+};
+
+const hasInvalidQuota = (quota: Quota): boolean => {
+  // Note: Invalid quota refers to negative quota received
+  if (quota === null) {
+    return true;
+  }
+  return quota.remainingQuota.some(item => item.quantity < 0);
+};
+
+export const useQuota = (
+  ids: string[],
+  authKey: string,
+  endpoint: string
+): QuotaHook => {
+  const [quotaState, setQuotaState] = useState<QuotaState>("DEFAULT");
+  const [quotaResponse, setQuotaResponse] = useState<Quota | null>(null);
+  const [allQuotaResponse, setAllQuotaResponse] = useState<Quota | null>(null);
+  const [quotaError, setQuotaError] = useState<Error>();
+  const { products } = useContext(ProductContext);
+  const prevIds = usePrevious(ids);
+  const prevProducts = usePrevious(products);
+
+  const updateQuota: QuotaHook["updateQuota"] = useCallback(() => {
+    const update = async (): Promise<void> => {
+      try {
+        setQuotaState("FETCHING_QUOTA");
+        const quota = await getQuota(ids, authKey, endpoint);
+        if (hasInvalidQuota(quota)) {
+          Sentry.captureException(
+            `Negative Quota Received: ${JSON.stringify(
+              quotaResponse?.remainingQuota
+            )}`
+          );
+          setQuotaState("NO_QUOTA");
+        } else if (hasNoQuota(quota)) {
+          setQuotaState("NO_QUOTA");
+        } else {
+          setQuotaState("DEFAULT");
+        }
+        setAllQuotaResponse(quota);
+        const filteredQuotas = filterQuotaWithAvailableProducts(
+          quota,
+          products
+        );
+        setQuotaResponse(filteredQuotas);
+      } catch (e) {
+        if (e instanceof NotEligibleError) {
+          setQuotaState("NOT_ELIGIBLE");
+          return;
+        } else {
+          Sentry.addBreadcrumb({
+            category: "useQuota",
+            message: "fetchQuota - unidentified error"
+          });
+          setQuotaState("DEFAULT");
+          setQuotaError(e);
+        }
+      }
+    };
+    update();
+  }, [ids, authKey, endpoint, products, quotaResponse]);
+
+  useEffect(() => {
+    if (prevIds !== ids || prevProducts !== products) {
+      updateQuota();
+    }
+  }, [prevIds, ids, prevProducts, products, updateQuota]);
```

Comment:
I noticed this hook automatically calls to get quota and also exposes the function so it can be called imperatively. I think we should add it in the comments of the `useQuota` hook so the developer who  uses it can know that it does both of these.

Reply:
I have added comments for this, let me know if they are sufficient.